dali/base/sysinfologger.cpp

```diff
@@ -0,0 +1,891 @@
+    }
+    virtual unsigned queryEndMonth() const override
+    {
+        return matchEndMonth;
+    }
+    virtual unsigned queryEndDay() const override
+    {
+        return matchEndDay;
+    }
+    virtual unsigned queryMatchYear() const override
+    {
+        return (matchStartYear==matchEndYear ? matchStartYear : 0);
+    }
+    virtual unsigned queryMatchMonth() const override
+    {
+        return (matchStartMonth==matchEndMonth ? matchStartMonth : 0);
+    }
+    virtual unsigned queryMatchDay() const override
+    {
+        return (matchStartDay==matchEndDay ? matchStartDay : 0);
+    }
+    virtual const char * queryMatchSource() const override
+    {
+        return matchSource.str();
+    }
+    virtual LogMsgCode queryMatchCode() const override
+    {
+        return matchCode;
+    }
+    virtual LogMsgAudience queryMatchAudience() const override
+    {
+        return matchAudience;
+    }
+    virtual LogMsgClass queryMatchClass() const override
+    {
+        return matchClass;
+    };
+    virtual unsigned __int64 queryMatchMsgId() const override
+    {
+        return matchMsgId;
+    };
+
+};
+
+ISysInfoLoggerMsgFilter * createSysInfoLoggerMsgFilter()
+{
+    return new CSysInfoLoggerMsgFilter();
+}
+
+class CSysInfoLoggerMsgIterator : public CInterface, implements ISysInfoLoggerMsgIterator
+{
+    Linked<ISysInfoLoggerMsgFilter> filter;
+    Owned<IRemoteConnection> conn = nullptr;
+    bool updateable = false;
+    Owned<IPropertyTree> root = nullptr;
+    Owned<IPropertyTreeIterator> msgIter = nullptr;
+    CSysInfoLoggerMsg infoMsg;
+
+    bool ensureMatch()
+    {
+        if (filter->hasDateRange())
+        {
+            for (; msgIter && msgIter->isValid(); msgIter->next())
+            {
+                unsigned __int64 ts = msgIter->query().getPropInt64(ATTR_TIMESTAMP, 0);
+                if(!ts || !filter->isInDateRange(ts))
+                    continue;
+                break;
+            }
+        }
+        return msgIter->isValid();
+    }
+
+public:
+    IMPLEMENT_IINTERFACE;
+
+    CSysInfoLoggerMsgIterator(ISysInfoLoggerMsgFilter * _filter, bool _updateable=false) : filter(_filter), updateable(_updateable)
+    {
+    }
+    CSysInfoLoggerMsg & queryInfoLoggerMsg()
+    {
+        return infoMsg.set(msgIter->get(), *(root.get()), updateable);
+    }
+    virtual ISysInfoLoggerMsg & query() override
+    {
+        return queryInfoLoggerMsg();
+    }
+    virtual bool first() override
+    {
+        StringBuffer xpath(SYS_INFO_ROOT);
+        if(filter->queryMatchYear() && filter->queryMatchMonth())
+        {
+            xpath.appendf("/m%04u%02u", filter->queryMatchYear(), filter->queryMatchMonth());
+            if (filter->queryMatchDay())
+                xpath.appendf("/d%02u", filter->queryMatchDay());
+        }
+        unsigned mode = RTM_LOCK_READ | (updateable?RTM_LOCK_WRITE : 0);
+        conn.setown(querySDS().connect(xpath.str(), myProcessSession(), mode, SDS_LOCK_TIMEOUT));
```

Comment:
there's a slight locking problem here, in that multiple clients/thread could connect to different points on the tree (multiple updates, multiple readers). e.g.
client1 : /SysLogs/m2024u12/d01 (reader)
client2 : /SysLogs/m2024u12 (writer)
etc.

i.e. they overlap, the locking will not protect each other.
I think you will always need to connect to SYS_INFO_ROOT to ensure writing doesn't clash with reading.